dags/common/utils.py

```diff
@@ -228,3 +227,63 @@ def process_archive(file_bytes, file_name):
         return process_zip_file(file_bytes, file_name)
     if tarfile.is_tarfile(file_bytes):
         return process_tar_file(file_bytes, file_name)
+
+
+def process_zip_file_elsevier(file_bytes, file_name):
+    file_bytes.seek(0)
+    with zipfile.ZipFile(file_bytes) as zip:
+        for filename in zip.namelist():
+            if "dataset.xml" not in filename:
+                continue
+            file_prefix = ".".join(file_name.split(".")[:-1])
+            zip_file_content = zip.read(filename)
+            s3_filename = os.path.join(file_prefix, filename)
+            return (zip_file_content, s3_filename)
+
+
+def process_tar_file_elsevier(file_bytes, file_name):
+    file_bytes.seek(0)
+    with tarfile.open(fileobj=file_bytes, mode="r") as tar:
+        for filename in tar.getnames():
+            if "dataset.xml" not in filename:
+                continue
+            file_prefix = ".".join(file_name.split(".")[:-1])
+            tar_file_content = tar.extractfile(filename).read()
+            s3_filename = os.path.join(file_prefix, filename)
+            return (tar_file_content, s3_filename)
+
+
+def process_archive_elsevier(file_bytes, file_name):
+    if zipfile.is_zipfile(file_bytes):
+        return process_zip_file_elsevier(file_bytes, file_name)
+    if tarfile.is_tarfile(file_bytes):
+        return process_tar_file_elsevier(file_bytes, file_name)
+
+
+def process_zip_file_specific_files(file_bytes, file_name, specific_file):
+    file_bytes.seek(0)
+    files = {}
+    with zipfile.ZipFile(file_bytes) as zip:
+        for filename in zip.namelist():
+            if filename == specific_file:
+                zip_file_content = zip.read(filename)
+                files[filename] = zip_file_content
+    return files
+
+
+def process_tar_file_specific_files(file_bytes, file_name, specific_file):
+    file_bytes.seek(0)
+    files = {}
+    with tarfile.open(fileobj=file_bytes, mode="r") as tar:
+        for filename in tar.getnames():
+            if filename in specific_file:
+                tar_file_content = tar.extractfile(filename).read()
+                files[filename] = tar_file_content
+    return files
+
+
+def process_archive_specific_files(file_bytes, file_name, specific_file):
```

Comment:
all these functions are just duplicates with an if statement, we don't need to do that, we can easily add a parameter to the original functions to process just the files we want.

Reply:
this one is different, it returns original file name, instead of the s3 key,